core/types/transaction.go

```diff
@@ -488,6 +503,47 @@ func (t *TransactionsByPriceAndNonce) CurrentSize() int {
 	return len(t.heads)
 }
 
+//Forward move t to be one index behind tx, tx cant be nil
+func (t *TransactionsByPriceAndNonce) Forward(tx *Transaction) {
+	if tx == nil {
+		txTmp := t.Peek()
+		for txTmp != nil {
+			t.Shift()
+			txTmp = t.Peek()
+		}
+		return
+	}
+
+	l := len(t.heads)
+	acc, _ := Sender(t.signer, tx)
```

Comment:
tx can be nil here because the miner worker is changing the value concurrently.

Reply:
Have went through the code, `worker` would pick `*tx` out from `txs(types.TransactionsByPriceAndNonce)` which eventually be nil, but not found nowhere modify `*tx`.